### PR TITLE
Add Stealing Artefacts plugin

### LINKED ISSUE #10348: Stealing Artefacts plugin
**Is your feature request related to a problem? Please describe.**
While stealing Artefacts in Port Piscarilius, it can be easy to forget which house your task was assigned to if not fully paying attention, if you go AFK mid-heist, or simply have bad memory. It can also be easy to forget where the pick-able drawers are for each location.

**Describe the solution you'd like**
**Feature 1:** An overlay in the corner that states which house the current heist is at after accepting a heist task (North-western, Western, South-Western, Northern, Southern, or South-eastern).
**Feature 2:** A highlight and arrow indicator for the drawers of the current heist. This could be the same highlight seen during clue steps that require drawer searching.
**Feature 3:** An auto NPC indicator highlight for the Patrolmen and Patrolwomen around the docks during a heist. This would make it easier to see which of the entities need to be avoiding while carrying the artefact.
**Feature 4:** Menu entry swap option for accepting a new task from Captain Khaled.

**Describe alternatives you've considered**
Currently I am using NPC Indicator with Patrolman and Patrolwoman as entries. For house location, I keep referencing [this image from the wiki](https://oldschool.runescape.wiki/images/c/c2/Piscarilius_residential_district_map.png?72707).

**Additional context**
Not all suggested features are dire. Even having just one or two of the four suggestions would be beneficial and a QoL improvement.

### PR DESCRIPTION
**Pull request by ThePharros**
Adds a plugin that assists in the Stealing Artefacts task in Kourend.

Closes #10348
Closes #10378

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsOverlay.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		setPosition(OverlayPosition.TOP_LEFT);
+		setPriority(OverlayPriority.LOW);
+		this.plugin = plugin;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+
+		panelComponent.getChildren().clear();
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsOverlay.java:59]: Clear the panel children after doing the region and state checks.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.stealingartefacts;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.api.*;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:32]: [Our code conventions are to avoid star imports.](https://github.com/runelite/runelite/wiki/Code-Conventions#imports)

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	private NPC captainKhaled = null;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:78]: `null` is the initial state of object fields, no need to declare it as such.
```suggestion
	private NPC captainKhaled;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:76]: Move constants to the top of the class, above injected fields.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() ->
+				stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+		}
+		else
+		{
+			stealingArtefactState = null;
+		}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:93]: Why is this here? `stealingArtefactState` should already null when the plugin starts.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	@Override
+	protected void shutDown() throws Exception
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:97]: This should also clear/null any other fields which maintain state, eg. `stealingArtefactState`, `captainKhaled`, etc.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() ->
+				stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:88]: This can be on one line. It'll be long, but it's still long as-is.
```suggestion
			clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 6 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:97]: This method doesn't throw any exceptions, so don't say it does.
```suggestion
	protected void shutDown()
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	private NPC captainKhaled = null;
+
+	@Override
+	protected void startUp() throws Exception
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:81]: This method doesn't throw any exceptions, so don't say it does.
```suggestion
	protected void startUp()
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		int stealingArtefactStateVarbit = client.getVar(Varbits.STEALING_ARTEFACT_STATE);
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:118]: This is only used once, inline it on the line below.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+		StealingArtefactState state = StealingArtefactState.getStealingArtefactState(stealingArtefactStateVarbit);
+
+		if (state != StealingArtefactState.DELIVER_ARTEFACT)
+		{
+			stealingArtefactState = state;
+			if (state != StealingArtefactState.NO_TASK && state != StealingArtefactState.FAILURE)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:124]: Compound if statements are easier to read if they're not negated. Swap the order of the blocks here:

```java
if (state == NO_TASK || state == FAILURE)
{
	client.clearHintArrow();
}
else
{
	client.setHintArrow(state.getWorldPoint());
}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+			}
+			else if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+			{
+				stealingArtefactState = StealingArtefactState.FAILURE;
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:160]: `NpcDespawned` also needs to be handled. Also, I believe NPCs are GC'd on game state change without firing despawn events? That must also be handled.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 5 above)
Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:99]: This is a remnant of honey's code. My guess is it may have been necessary with the previous code as there was no null on shut down, which there now will be.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 11 above)
Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:168]: Alright I've just added `onNpcDespawn()` handling. What is GC in this context?

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+	LOCATION_WEST("West House", Color.YELLOW, new WorldPoint(1747, 3749, 1)),
+	LOCATION_NORTHWEST("Northwest House", Color.YELLOW, new WorldPoint(1750, 3763, 1)),
+	FAILURE("Failure", Color.RED, new WorldPoint(0, 0, 0)),
+	DELIVER_ARTEFACT("Deliver Artefact", Color.GREEN, new WorldPoint(0, 0, 0));
+
+	private final String description;
+	private final Color color;
+	private static final Map<Integer, StealingArtefactState> stealingArtefactStates;
+
+	private WorldPoint worldPoint;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:55]: Switch the positions of these. Fields and static constants shouldn't be mixed. Above should read:

```java
private final String description;
private final Color color;
private final WorldPoint worldPoint;

private static final Map<Integer, StealingArtefactState> stealingArtefactStates;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+		for (StealingArtefactState stealingArtefactState : values())
+		{
+			builder.put(stealingArtefactState.ordinal(), stealingArtefactState);
+		}
+
+		stealingArtefactStates = builder.build();
+	}
+
+	static StealingArtefactState getStealingArtefactState(int id)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:69]: This method name is rather verbose, and its argument name a bit vague. I'd recommend the following.
```suggestion
	static StealingArtefactState getState(int varbitValue)
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:84]: Lombok's generated getter will automatically name this field's getter method `isInPortPiscariliusRegion` since it is a boolean; the field's name should not start with `is`.
```suggestion
	private boolean inPortPiscariliusRegion;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 11 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:166]: Garbage collected. You need to reset any stored NPCs on game state change:

```java
@Subscribe
public void onGameStateChanged(GameStateChanged event)
{
	if (event.getGameState() != LOGGED_IN)
	{
		captainKhaled = null;
	}
}
```

**Comment by ThePharros** [comment]
Came across a minor issue:

The hint arrow does not get cleared when leaving the region, however it is only for local space render and not worldmap render, so it is only noticeable when you are just outside the region when the region is still loaded.

I can’t clear the hintarrow on game tick when outside the region since it would negate other hint arrows. Should this be looked into? Any ideas on tackling it?

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean inPortPiscariliusRegion;
+
+	private NPC captainKhaled;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:86]: Again, you need to null this field when game state changes since that will garbage collect NPCs without firing a despawn event.

Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:86]: as in just add:
`    @Subscribe
	public void onGameStateChanged(GameStateChanged event)
	{
		captainKhaled = null;
	}`
that?

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 16]
```diff
@@ -53,149 +56,167 @@
...
+            }
+        }
+    }
+
+    @Subscribe
+	public void onGameStateChanged(GameStateChanged event)
 	{
-		client.clearHintArrow();
-		stealingArtefactState = null;
 		captainKhaled = null;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:171]: I guess don't clear it if the state is LOGGED_IN. Otherwise, this should be fine.
```suggestion
		if (event.getGameState() != GameState.LOGGED_IN)
		{
			captainKhaled = null;
		}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 17]
```diff
@@ -26,12 +26,15 @@
 package net.runelite.client.plugins.stealingartefacts;
 
 import com.google.common.collect.ImmutableSet;
+
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:29]: Remove the blank lines added into the import block

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsOverlay.java [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+		this.plugin = plugin;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+
+		StealingArtefactState stealingArtefactState = plugin.getStealingArtefactState();
+
+		if (!plugin.isInPortPiscariliusRegion() || stealingArtefactState == null)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsOverlay.java:58]: When using the north escape (for the west and southwest houses), you wander slightly outside the defined port piscarilius regions. It would be nice to continue to show the overlay in that case:
```suggestion
		if (stealingArtefactState == null || (stealingArtefactState != StealingArtefactState.DELIVER_ARTEFACT && !plugin.isInPortPiscariliusRegion()))
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+	LOCATION_SOUTH("South House", Color.YELLOW, new WorldPoint(1764, 3735, 1)),
+	LOCATION_SOUTHWEST("Southwest House", Color.YELLOW, new WorldPoint(1749, 3735, 1)),
+	LOCATION_WEST("West House", Color.YELLOW, new WorldPoint(1747, 3749, 1)),
+	LOCATION_NORTHWEST("Northwest House", Color.YELLOW, new WorldPoint(1750, 3763, 1)),
+	FAILURE("Failure", Color.RED, new WorldPoint(0, 0, 0)),
+	DELIVER_ARTEFACT("Deliver Artefact", Color.GREEN, new WorldPoint(0, 0, 0));
+
+	private final String description;
+	private final Color color;
+	private WorldPoint worldPoint;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:53]:
```suggestion
	private final WorldPoint worldPoint;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+	private static final Map<Integer, StealingArtefactState> stealingArtefactStates;
+
+	static
+	{
+		ImmutableMap.Builder<Integer, StealingArtefactState> builder = new ImmutableMap.Builder<>();
+
+		for (StealingArtefactState stealingArtefactState : values())
+		{
+			builder.put(stealingArtefactState.ordinal(), stealingArtefactState);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:63]: A map indexed by enum ordinal isn't necessary, you can just use `values()` to get an array and index that.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 21]
```diff
@@ -0,0 +1,221 @@
...
+				client.setHintArrow(stealingArtefactState.getWorldPoint());
+			}
+		}
+		else if (captainKhaled != null)
+		{
+			client.setHintArrow(captainKhaled);
+		}
+		else
+		{
+			client.clearHintArrow();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:144]: This should probably not clear the hint arrow unless it is more-sure the current arrow is its own. A lot of content has hint arrows.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 22]
```diff
@@ -0,0 +1,221 @@
...
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		if (event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:151]: The event has the inventory id in it, you can just check that.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 23]
```diff
@@ -0,0 +1,221 @@
...
+		{
+			if (hasArtefact())
+			{
+				stealingArtefactState = StealingArtefactState.DELIVER_ARTEFACT;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:155]: I don't understand how this is necessary if the state is based off of the var. Wouldn't the var change just do this?

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 24]
```diff
@@ -0,0 +1,221 @@
...
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGGED_IN)
+		{
+			captainKhaled = null;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:169]: npc despawn events still fire even on scene changes, so this probably isn't necessary

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 25]
```diff
@@ -0,0 +1,221 @@
...
+			{
+				client.setHintArrow(captainKhaled);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		if (event.getNpc().getId() == NpcID.CAPTAIN_KHALED_6972)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:189]: This can just be a reference comparison with the stored npc.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 21 above)
Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:144]: does that go for the entire method's hint arrow clearing or just the one in the else statement? For example, if the varbit changes to NO_TASK or FAILURE, I would think they the player is actively doing SA, as to get FAILURE the player would had have the artefact and were either caught or left the region, and to get NO_TASK they would need to speak to Khaled after a failed delivery.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 23 above)
Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:155]: you're right, looks to be redundant. i'll remove it.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 24 above)
Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:169]: does that extend to world hopping and/or disconnects?

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 21 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:144]: Anytime the plugin might clear the hint arrow, it should check to make sure the player is actually interacting with (or just stopped interacting with) the minigame.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java (hunk 19 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:53]: this isn't resolved

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java (hunk 20 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactState.java:63]: this isn't resolved

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 25 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:189]: this isn't resolved

Comment (ThePharros) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:189]: ive been resolving convos as i go down in my own commit prior to push. should i switch to only doing that post push then?

Comment (Spedwards) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:189]: Resolve them once they've been resolved... That means pushed here so we can see that it's been resolved.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 26]
```diff
@@ -0,0 +1,167 @@
...
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.values()[client.getVar(Varbits.STEALING_ARTEFACT_STATE)]);
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		client.clearHintArrow();
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:93]: This should do the same region check as is done for the game tick clearing of hint arrows.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java (hunk 21 above)
Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:111]: This does not appear to be resolved.

File: runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java [hunk 27]
```diff
@@ -0,0 +1,167 @@
...
+			{
+				client.setHintArrow(captainKhaled);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		if (captainKhaled != null && event.getNpc().getId() == captainKhaled.getId())
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java:135]: This is not resolved. Per Adam's comment:

> This can just be a reference comparison with the stored npc.

```suggestion
		if (event.getNpc() == captainKhaled)
```

**Comment by samcaulfield** [comment]
Hi, FYI I just play-tested this plugin for an hour, including lots of teleporting in/out of the area and found it quite useful (and didn't encounter any issues with it). Thanks for writing it @ThePharros!

I see that there are some outstanding changes before this PR can be merged and I am willing to help implement them if the original authors don't have the time to.

**Comment by Nightfirecat** [comment]
Given that we have a hub plugin available for this now, and it's rather niche content, I feel this is no longer needed as a main client plugin. I recommend contributing to the hub plugin or contributing a separate hub plugin if you're interested.
